corehq/apps/locations/forms.py

```diff
@@ -654,9 +654,15 @@ def save(self):
         selected_location_ids = set(self.cleaned_data['related_locations'].split(','))
         previous_locations = self.location.related_location_ids
         locations_to_add = selected_location_ids - previous_locations
+        locations_to_remove = previous_locations - selected_location_ids
 
         for location_id in locations_to_add:
             LocationGroup.objects.get_or_create(
                 location_a=self.location,
                 location_b=SQLLocation.objects.get(location_id=location_id)
             )
+
+        LocationGroup.objects.filter(
+            location_a=self.location, location_b__location_id__in=locations_to_remove).delete()
+        LocationGroup.objects.filter(
+            location_b=self.location, location_a__location_id__in=locations_to_remove).delete()
```

Comment:
Since you specified "location_id" as the field to use in defining these relationships, do you know if
`location_a_id__in` would work instead to avoid the join?  (Or would it be `location_a_location_id__in`)?

Reply:
eh, I guess in this context performance probably wouldn't ever be a concern.  I'm still curious, but feel free to disregard.